flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraph.java

```diff
@@ -153,6 +159,8 @@ public ArchivedExecutionGraph(
         this.checkpointStorageName = checkpointStorageName;
         this.stateChangelogEnabled = stateChangelogEnabled;
         this.changelogStorageName = changelogStorageName;
+        this.streamGraphJson = streamGraphJson;
```

Comment:
checkNotNull

Reply:
Could we set the streamGraphJson to null in the JobDetailsInfo if the JSON is not valid, regardless of whether it is an empty string or an empty JSON object?